src/test/java/htsjdk/samtools/SeekableByteChannelFromBuffer.java

```diff
@@ -77,6 +77,10 @@ public void close() throws IOException {
     open = false;
   }
 
+  protected ByteBuffer getBuffer(){
```

Comment:
Seems a shame to expose this for testing purposes only.  If the subclass type was defined in the same package, you could use package (default) scope instead, which is more restrictive than protected, and doesn't lead to API compatibility requirements later on.

So you could use default scope here if you defined `OneByteAtATimeChannel` in the same package as `SeekableByteChannelFromBuffer`. You could put the source in the test directory so it wouldn't be part of the public API.

Reply:
This class is already in the test directories so it isn't part of the public API.  I don't mind moving them both to a subpackage though and restricting it to default scope.

Reply:
moved both to htsjdk.testutil.streams